graphql/resolvers.go

```diff
@@ -1850,10 +1851,48 @@ func (r *queryResolver) TaskTestSample(ctx context.Context, tasks []string, test
 			}
 			testResultsToReturn = append(testResultsToReturn, tr)
 		}
-		return testResultsToReturn, nil
+	} else {
+		regexFilter := ""
+		for i, f := range testFilters {
+			regexFilter += f.TestName
+			// If there is another test filter add a pipe to separate them
+			if i < len(testFilters)-1 {
+				regexFilter += "|"
+			}
+		}
+		for _, t := range t {
+			filteredTestResults, err := r.sc.FindTestsByTaskId(data.FindTestsByTaskIdOpts{
+				TaskID:    t.Id,
+				Execution: t.Execution,
+				TestName:  regexFilter,
+				Statuses:  []string{evergreen.TestFailedStatus},
+				Limit:     10,
+				Page:      0,
+			})
+			if err != nil {
+				return nil, ResourceNotFound.Send(ctx, err.Error())
```

Comment:
Shouldn't this be InternalServerError? I don't think it should be an error if no tests are found.